components/eventing-controller/controllers/subscription/nats/integration_test.go

```diff
@@ -0,0 +1,837 @@
+					reconcilertesting.HaveCondition(
+						conditionInvalidSink("sink does not contain suffix: svc.cluster.local in the URL")),
+				},
+				k8sEvents: []v1.Event{
+					eventInvalidSink("Sink does not contain suffix: svc.cluster.local")},
+			},
+		},
+
+		{
+			name: "invalid sink; too many sub domains",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURL(fmt.Sprintf("https://%s.%s.%s.svc.cluster.local", "testapp", "testsub", "test")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(
+						conditionInvalidSink("sink should contain 5 sub-domains: testapp.testsub.test.svc.cluster.local")),
+				},
+				k8sEvents: []v1.Event{
+					eventInvalidSink("Sink should contain 5 sub-domains: testapp.testsub.test.svc.cluster.local")},
+			},
+		},
+
+		{
+			name: "invalid sink; wrong namespace",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURL(fmt.Sprintf("https://%s.%s.svc.cluster.local", "testapp", "wrong-ns")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(
+						conditionInvalidSink("namespace of subscription: test and the namespace of subscriber: wrong-ns are different")),
+				},
+				k8sEvents: []v1.Event{
+					eventInvalidSink("natsNamespace of subscription: test and the subscriber: wrong-ns are different")},
+			},
+		},
+
+		{
+			name: "invalid sink; not a valid cluster local service",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURL(
+					reconcilertesting.ValidSinkURL(ens.subscriberSvc.Namespace, "testapp")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(
+						conditionInvalidSink("sink is not valid cluster local svc, failed with error: Service \"testapp\" not found")),
+				},
+				k8sEvents: []v1.Event{
+					eventInvalidSink("Sink does not correspond to a valid cluster local svc")},
+			},
+		},
+
+		{
+			name: "valid sink",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens)),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
```

Comment:
nit: `reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition(""))` does also test that `reconcilertesting.HaveSubscriptionReady()`.

We should use it consistently (either always both statements or only the first statement). There are some calls to `		reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),` without `reconcilertesting.HaveSubscriptionReady(),`

An idea from another comment is to rename `DefaultCondition ` to `DefaultReadyCondition`. This covers both lines.

```suggestion
					reconcilertesting.HaveCondition(reconcilertesting.DefaultReadyCondition("")),
```

Reply:
true, removed the Ready and renamed to `DefaultReadyCondition `